### ISSUE TITLE
🧪 Test Hedy on the OLPC browser

### ISSUE THREAD

**Issue by Felienne** [hedyorg/hedy#5207]
Recently I learned that the One Laptop Per Child project has its own custom software including a browser: https://en.wikipedia.org/wiki/OLPC_XO#Software

Let's try to see if Hedy works well on that!

This is in principle an easy issue, no knowledge of Hedy is required

**Comment by El-Houssein-bedoui** [comment]
Would using a virtual machine with OLPC-like specifications be adequate to test Hedy on the OLPC browser? By replicating the hardware environment, could we effectively evaluate compatibility?

**Comment by Felienne** [comment]
> Would using a virtual machine with OLPC-like specifications be adequate to test Hedy on the OLPC browser?

Yes, I think so! I think a great step one is to Google and figure out what type of browser it even is so we can inform further research. I have done no inquiry at all yet.

**Comment by El-Houssein-bedoui** [comment]
The OLPC XO-1 and its early iterations were primarily designed for educational purposes, offering limited web browsing capabilities within the Sugar environment. You can  try the sugar environment of the OLPC XO-1 using this website : https://server.sugarizer.org

Picture of the sugar site (**does not contain web browsing**):
![image](https://github.com/hedyorg/hedy/assets/141435410/8702329f-40fd-40f6-af3e-4251c64bc705)

While the OLPC XO-4, especially with its touchscreen functionality, provided enhanced support for web browsing. Despite attempting to download and simulate an OLPC XO-4 image using VMware Player, I encountered challenges as only OLPC-XO-1 images were **available** and the provided instructions were **outdated.**

Picture of the outdated instruction ( **insufficient for practical purpose**):
![image](https://github.com/hedyorg/hedy/assets/141435410/bee484f7-ab34-4cf7-bc4d-c7c5d2df0408)

Considering these limitations, the most pragmatic approach seems to be testing the website on a physical OLPC XO-4 device. Thank you.

**Comment by yo-sh-i** [comment]
Hi @Felienne en @El-Houssein-bedoui , look at this ;-) . 

<img width="578" alt="Schermafbeelding 2024-10-02 150249" src="https://github.com/user-attachments/assets/0e691784-a840-4f2b-bcac-dd27fa944bc6">

<img width="578" alt="Schermafbeelding 2024-10-02 150323" src="https://github.com/user-attachments/assets/e585b892-959f-459c-b421-bf7fa19be73f">

I will post the instructions in another comment to get this working for yourself if you would like to. What I immediately noticed is that the emoji's don't render. But maybe if we will replace emoji's with fontawesome icons as per #5100 this won't be an issue anymore in the future.

**Comment by yo-sh-i** [comment]
Some facts as per the wiki of OLPC: https://wiki.laptop.org/go/The_OLPC_Wiki

The OLPC laptops come in different models with different system software installed from the factory.

- XO-1, XO-1.5, XO-1.75 and XO-4 laptops run a modified Fedora linux distro. On top of that you will find the "Sugar" user interface, but also a GNOME desktop environment is provided.
- Model XO-4 comes in two flavours; a non-touch and a touch variant. Optionally Android can be installed.
- Besides the above four laptops, there is the NL3 laptop. This is based on Ubuntu 16.04 LTS.

What I found interesting is that the most recent version of the system software (13.2.11) of the XO-laptops is based on Fedora 18 (from 2016!) as per https://wiki.laptop.org/go/Releases and that the included browser, indeed called Browse, is based on a modified Firefox version, more specific the underlying browser engine of Firefox XULrunner/Gecko.

However, the above two screenshots are from the Sugar on a Stick release (https://wiki.sugarlabs.org/go/Sugar_on_a_Stick). This is an easy way of working with the Sugar UI based on more recent Fedora versions. You can find the instructions to install it with e.g. VirtualBox or VMWare Workstation on the link above. The exact browser finger print can be found seen in the below screenshot. 

**So, my conclusion is that my tests aren't representable for the actual system software provided for the XO-laptops, rather it gives some idea how Hedy runs on Fedora v40 > Browse v208 with the below browser fingerprint data.**

![Schermafbeelding 2024-09-24 180012](https://github.com/user-attachments/assets/fc48db43-aded-4a31-b049-15ddd869091f)

**Comment by brianpeiris** [comment]
There's more info about Sugar here: https://wiki.laptop.org/go/Sugar
Sugar is maintained by Sugar Labs, and they provide a bootable ISO image based on Fedora Linux that you can run on compatible hardware and emulators. https://www.sugarlabs.org/try-sugar https://fedoraproject.org/spins/soas/download

I was able to get it running in VirtualBox, and it seems the Browse application is based on Webkit 605 (equivalent to Apple's Safari 11 browser), which was released around 2018

Hedy seems to work well in basic tests at least:

<img width="1448" height="1125" alt="Image" src="https://github.com/user-attachments/assets/f282aefe-bc49-4400-9a8f-f06e68d167d5" />
